Update CHANGELOG.md for 4.72.0 (#5267)
AzureAD/microsoft-authentication-library-for-dotnet: CHANGELOG.md

4.72.0
=======
### Features
- Added MacOs Broker support. See [Issue #5051](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5051)

> - Retry policies for ManagedIdentities are now per request instead of per HttpManager [Issue #5252](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5252)

Reviewer: ```suggestion
- Retry policies for Managed Identity is now per request instead of per HttpManager [Issue #5252](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5252)
```

Reviewer 2: I would remove this compelely or add it to a "engineering improvements". There is not feature update. HttpManager is not a customer-facing API

Author: Oh I see, didn't realize it was internal only. I can remove.
